iocore/net/quic/QUICTLS.cc

```diff
@@ -91,44 +89,69 @@ QUICTLS::is_ready_to_derive() const
   }
 }
 
-bool
-QUICTLS::is_key_derived(QUICKeyPhase key_phase, bool for_encryption) const
-{
-  if (key_phase == QUICKeyPhase::ZERO_RTT) {
-    return this->_client_pp->get_key(QUICKeyPhase::ZERO_RTT);
-  } else {
-    return this->_get_km(key_phase, for_encryption);
-  }
-}
-
 int
 QUICTLS::initialize_key_materials(QUICConnectionId cid)
 {
+  this->_pp_key_info.set_cipher_initial(EVP_aes_128_gcm());
+  this->_pp_key_info.set_cipher_for_hp_initial(EVP_aes_128_ecb());
+
   // Generate keys
   Debug(tag, "Generating %s keys", QUICDebugNames::key_phase(QUICKeyPhase::INITIAL));
-  std::unique_ptr<KeyMaterial> km;
 
-  km = this->_keygen_for_client.generate(cid);
-  this->_print_km("initial - client", *km);
-  this->_client_pp->set_key(std::move(km), QUICKeyPhase::INITIAL);
+  uint8_t *client_key_for_hp;
+  uint8_t *client_key;
+  uint8_t *client_iv;
+  size_t client_key_for_hp_len;
+  size_t client_key_len;
+  size_t *client_iv_len;
 
-  km = this->_keygen_for_server.generate(cid);
-  this->_print_km("initial - server", *km);
-  this->_server_pp->set_key(std::move(km), QUICKeyPhase::INITIAL);
+  uint8_t *server_key_for_hp;
+  uint8_t *server_key;
+  uint8_t *server_iv;
+  size_t server_key_for_hp_len;
+  size_t server_key_len;
+  size_t *server_iv_len;
 
-  return 1;
-}
+  if (this->_netvc_context == NET_VCONNECTION_IN) {
+    client_key_for_hp     = this->_pp_key_info.decryption_key_for_hp(QUICKeyPhase::INITIAL);
+    client_key_for_hp_len = this->_pp_key_info.decryption_key_for_hp_len(QUICKeyPhase::INITIAL);
```

Comment:
don't understand when and where to set `_client_key_for_hp_len` in `pp_key_info` and what's value of client_key_for_hp_len  on this line.

Reply:
never mind . I found it